driver-cql-shaded/src/main/java/io/nosqlbench/activitytype/cql/datamappers/functions/to_daterange/DateRangeFunc.java

```diff
@@ -75,13 +75,12 @@ public DateRangeFunc(String precision, Function<Long,Long> lower, LongUnaryOpera
         this.upper = upper;
     }
 
-
     @Override
     public DateRange apply(long value) {
         Date lowerDate = new Date(lower.applyAsLong(value));
         DateRange.DateRangeBound lower = DateRange.DateRangeBound.lowerBound(lowerDate,precision);
         Date upperDate = new Date(upper.applyAsLong(value));
-        DateRange.DateRangeBound upper = DateRange.DateRangeBound.upperBound(lowerDate,precision);
+        DateRange.DateRangeBound upper = DateRange.DateRangeBound.upperBound(upperDate,precision);
```

Comment:
did not work before and might change existing workloads?

Reply:
As it could change existing workloads, maybe we should add a call out in release notes for it. However, I do believe the best thing to do is make it correct.